### ISSUE TITLE
[Meta Discussion]: I will no longer be submitting many (any?) more error reports.

### ISSUE THREAD

**Issue by WraithGlade** [Apress/practical-shader-dev#23]
It is clear at this point that this book has a lot of editorial problems with it and many miscellaneous inconsistencies throughout the entire text and the accompanying GitHub code. Probably there were multiple revisions of the text and code, but the author and/or publisher didn't properly synchronize those changes and account for them. Some areas also lack precision and clarity somewhat as well (e.g. some fairly poor variable names, occasional subtly misleading explanations, some things said later in the book that don't match with things said previously, etc).

It is taking up a lot of time to report these issues, and I would rather use my time for more productive things right now. It also appears that the author/publisher of this book doesn't attend to this GitHub account much, which contributes to demotivating additional reporting, since it may never be used potentially.

**However, just because I am not submitting many/any more issues for this book doesn't mean they won't still be there. They almost certainly will be, if past experiences so far is any indication.**

**You should do a new editing pass through the entire book and also the posted example code and clean up any inconsistencies.**

As much as I very often feel strongly compelled by my sense of duty to report issues with books or code projects as best I can, I have decided to nonetheless resist my compulsion for diligent reporting this time, for the remainder of this book, in the interest of my own time and productivity.

If I report anything else here after this, it will only be out of my own generosity or to inquire about a critical issue etc.

**Comment by hamoid** [comment]
I agree with the need of a new editing pass. I came to this site trying to find an online page listing typos and addendum, but it does not exist right?

I find the book is full of good concepts and explanations, but unfortunately also full of small mistakes.  Would be nice if they gave us the money back for reporting 25 issues with it.

**Comment by bferguson3** [comment]
Thank you for your submissions. I think the book is fairly well written for my skill level and Halladay manages to squeeze some good tips in there, but I spend more time than I should debugging their own code.

Maybe we'll get a second edition.
